crates/client/rpc/src/lib.rs

```diff
@@ -1580,47 +1645,24 @@ where
             events_converted.last().unwrap().data[2]
         };
 
-        let messages = self
-            .client
-            .runtime_api()
-            .get_tx_messages_to_l1(substrate_block_hash, Felt252Wrapper(transaction_hash).into())
-            .map_err(|e| {
-                error!("'{e}'");
-                StarknetRpcApiError::InternalServerError
-            })?;
+        let messages = self.get_tx_messages_to_l1(substrate_block_hash, transaction_hash)?;
 
-        fn message_conversion(message: starknet_api::transaction::MessageToL1) -> starknet_core::types::MsgToL1 {
-            let mut to_address = [0u8; 32];
-            to_address[12..32].copy_from_slice(message.to_address.0.as_bytes());
-            starknet_core::types::MsgToL1 {
-                from_address: Felt252Wrapper::from(message.from_address).0,
-                to_address: FieldElement::from_bytes_be(&to_address).unwrap(),
-                payload: message.payload.0.into_iter().map(|felt| Felt252Wrapper::from(felt).0).collect(),
-            }
-        }
+        let messages_sent = messages.into_iter().map(message_conversion).collect();
+
+        // TODO: use actual execution resources
+        let execution_resources = get_execution_resources();
```

Comment:
You don't want to do in this PR?

Reply:
Unfortunately, I have no idea how to get it... I can do that if you @tdelabro have any thoughts.